test/AlpineLinux/ParserSpec.hs

```diff
@@ -0,0 +1,70 @@
+module AlpineLinux.ParserSpec (spec) where
+
+import Data.Text (Text)
+import Data.Text.IO qualified as TIO
+import Strategy.AlpineLinux.Parser (PackageError (..), installedPackagesDatabaseParser)
+import Strategy.AlpineLinux.Types (AlpinePackage (..))
+import Test.Hspec (Spec, SpecWith, before, describe, expectationFailure, it, shouldBe)
+import Text.Megaparsec (parse)
+
+testFilePath :: String
+testFilePath = "test/AlpineLinux/testdata/installed.example"
+
+testFileName :: String
+testFileName = "installed.example"
+
+spec :: Spec
+spec =
+  describe "installedPackagesDatabaseParser" $ do
+    it "parses empty files" $ do
+      let packages = parse installedPackagesDatabaseParser testFileName ""
+      case packages of
+        Left err ->
+          expectationFailure $ "Unexpected parse failure: " <> show err
+        Right val ->
+          val `shouldBe` []
+    withExampleDatabase $
+      it "parses the example database" $ \exampleDatabase -> do
+        let packages = parse installedPackagesDatabaseParser testFileName exampleDatabase
+        case packages of
+          Left err ->
+            expectationFailure $ "Unexpected parse failure: " <> show err
+          Right p -> do
+            length p `shouldBe` 14
+            (take 1 . drop 5 $ p)
```

Comment:
[Nit, Optional] We should assert all of the entries in the file, or not include the other entries in the file

Reply:
I went back and forth on it.  It was a lot to write manually when I was making the test before the code was done.  But now that I have things working it should be easy enough to run it, see the output, double-check it and copy it into the test file.  That's a lot less work than building all the objects manually.